sqlmesh/core/model/definition.py

```diff
@@ -622,7 +619,9 @@ def depends_on_past(self) -> bool:
             query = self.render_query(optimize=False)
             if query is None:
                 return False
-            return self.name in d.find_tables(query, dialect=self.dialect)
+            return self.fqn in d.find_tables(
```

Comment:
remove the depends_on_past cache and the depends_on cache. instead, cache the result of find_tables and use that on depends_on_past and depends_on